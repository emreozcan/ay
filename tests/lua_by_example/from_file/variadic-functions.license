https://luabyexample.techplexlabs.com/variadic-functions/
Lua by Example: Variadic Functions
by Blake Bourque

This work is licensed under a Creative Commons Attribution 3.0 Unported License.