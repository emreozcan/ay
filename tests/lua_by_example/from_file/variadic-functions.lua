local function sum(...)
    print("items: ", ...)
    total = 0
    for idx, num in ipairs({...}) do
        total = total + num
    end
    print("total: ",total)
end

sum(1, 2)
sum(1, 2, 3)

nums = {1, 2, 3, 4}
sum(table.unpack(nums))