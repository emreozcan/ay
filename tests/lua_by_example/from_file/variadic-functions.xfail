This test uses table.unpack(), which we do not have yet.
