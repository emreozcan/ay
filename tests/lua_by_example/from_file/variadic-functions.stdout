items: 	1	2
total: 	3
items: 	1	2	3
total: 	6
items: 	1	2	3	4
total: 	10
